pyanaconda/modules/common/base/base.py

```diff
@@ -222,11 +223,31 @@ def read_kickstart(self, s):
         log.debug("Reading kickstart...")
         report = KickstartReport()
 
+        ks_module = "pykickstart"
+
+        def log_ks_warning(message, category, filename, lineno, file=None, line=None):
+            """Collect pykickstart warnings.
+
+            Based on pyanaconda.kickstart.parseKickstart.ksshowwarning()
+            """
+            if ks_module in filename and issubclass(category, KickstartParseWarning):
+                data = KickstartMessage.for_warning(str(message))
+                report.warning_messages.append(data)
+            # No need to print the warning here, the UI will take care of that
+
         try:
             handler = self.get_kickstart_handler()
             parser = self.get_kickstart_parser(handler)
-            parser.readKickstartFromString(s)
-            self.process_kickstart(handler)
+
+            with warnings.catch_warnings():
```

Comment:
I am pretty sure that we can use `warnings.catch_warnings(record=True)` here to avoid the function `log_ks_warning`.